enclave/src/openfinex/response_handler.rs

```diff
@@ -114,8 +115,12 @@ impl PolkadexResponseHandler {
         }
     }
 
-    fn handle_trade_event(&self, _trade_event: TradeEvent) {
+    fn handle_trade_event(&self, trade_event: TradeEvent) {
         debug!("Received trade event from OpenFinex");
+        if let Err(e) = polkadex_gateway::settle_trade(trade_event) {
+            error!("[Error] in polkadex gateway settle trade: {:?}", e);
+        };
```

Comment:
`settle_trade` should be part of `PolkaDexGatewayCallback` and then be called through the trait, rather than calling a static function. Otherwise it will strongly couple the response handler to the gateway again.

Reply:
It is not a Callback though?

Reply:
Still, I agree we should decouple it and I'll place it in the Callback, but maybe we should rename it.

Reply:
Callback is just a name, if it doesn't fit anymore, replace it with a better one 😄 The important thing is to keep it de-coupled (and hence testable) through the trait.